test/unit/annotation_spec.js

```diff
@@ -546,16 +552,16 @@ describe('annotation', function() {
         docBaseUrl: 'http://www.example.com/test/pdfs/qwerty.pdf',
       });
 
-      var annotation = AnnotationFactory.create(xref, annotationRef,
-                                                pdfManager, idFactoryMock);
-      var data = annotation.data;
-      expect(data.annotationType).toEqual(AnnotationType.LINK);
-
-      expect(data.url).toEqual(
-        new URL('http://www.example.com/test/pdfs/Part II/Part II.pdf').href);
-      expect(data.unsafeUrl).toEqual('Part II/Part II.pdf');
-      expect(data.dest).toBeUndefined();
-      expect(data.newWindow).toEqual(true);
+      AnnotationFactory.create(xref, annotationRef,
+                pdfManager, idFactoryMock).then((annotation) => {
```

Comment:
Indent this like you did here: https://github.com/mozilla/pdf.js/pull/9417/files#diff-04ad2b4c022ab1eae0d58931b8fe0b7dR74